support-frontend/assets/components/contributionsReminder/contributionsReminder.jsx

```diff
@@ -0,0 +1,162 @@
+// @flow
+
+// ----- Imports ----- //
+
+import React, { Component } from 'react';
+import SvgSubscribe from 'components/svgs/subscribe';
+import SvgSubscribed from 'components/svgs/subscribed';
+import TrackableButton from 'components/button/trackableButton';
+import { trackComponentClick, trackComponentLoad } from 'helpers/tracking/behaviour';
+import { connect } from 'react-redux';
+import type { Csrf as CsrfState } from 'helpers/csrf/csrfReducer';
+import { routes } from 'helpers/routes';
+import { logException } from 'helpers/logger';
+
+// ----- Types ----- //
+type PropTypes = {
+  email: string | null,
+  csrf: CsrfState,
+}
+
+type StateTypes = {
+  clicked: boolean,
+  pending: boolean,
+  successful: boolean,
+}
+
+const mapStateToProps = state => ({
+  email: state.page.form.formData.email,
+  csrf: state.page.csrf,
+});
+
+// ----- Render ----- //
+class ContributionsReminder extends Component<PropTypes, StateTypes> {
+  static defaultProps = {
+    email: null,
+  }
+
+  state = {
+    clicked: false,
+    pending: false,
+    successful: false,
+  }
+
+  setButtonClickedState = (): void => {
+    this.setState({
+      clicked: true,
+      pending: true,
+    });
+  }
+
+  setSuccessfulState = (): void => {
+    this.setState({
+      pending: false,
+      successful: true,
+    });
+  }
+
+  setFailedState = (): void => {
+    this.setState({
+      pending: false,
+      successful: false,
+    });
+  }
+
+  renderButtonCopy = (): string => {
+    const { clicked, pending, successful } = this.state;
+    const defaultState = !clicked && !pending && !successful;
+    const pendingState = clicked && pending && !successful;
+    const failedState = clicked && !pending && !successful;
+    const successState = clicked && !pending && successful;
```

Comment:
I realize these are overly verbose-- eg default could be just `!clicked` -- but I think this approach makes them more readable/debuggable

Reply:
I think the simplest way to model this would be:
```
type State = 'Initial' | 'Pending' | 'Failed' | 'Successful'
...

state = 'Initial'
...

switch (state) {
  case 'Initial':
    return 'Remind me in December';
  case 'Pending':
    ...
}

```

Reply:
I like demonstrating that the three client events add up to a state but i can refactor if you think your way is more maintainable

Reply:
I think each client event would still map to a distinct state, e.g.
```
onButtonClicked = (): void => {
  this.setState('Pending');
}
```

Reply:
i'm not 100% sure about this but i think state has to be an object